### ISSUE TITLE
Stuck on "Loading..." in version 1.6.0

### ISSUE THREAD

**Issue by lucasbacchi** [lbell/pretty-google-calendar#28]
Hi there! Thank you so much for your development work here!

## Background
The plugin automatically updated a few days ago and suddenly the embed wouldn't load.

This is how the embed renders:
![image](https://github.com/lbell/pretty-google-calendar/assets/43394783/739c4fac-90b0-4d30-b741-db965823ca9c)
It appears to be stuck on loading and never receiving a response from the API (or never sending one... More on that later).

## Troubleshooting
I started troubleshooting by looking at the logs in Google Cloud. It showed that there hadn't been any requests to the Google Calendar API in the last few days. Interestingly, there were also no errors logged. There wasn't anything in the browser console, so I didn't have much to go on, but I tried clearing caches and that didn't help. This is about when I discovered that a new version had just been released. v1.4.1 was the last version I remember having so I decided to roll back. As soon as I loaded that version back onto the server, the calendar started working again.

I've taken a look at the diff from [v1.4.1 to v1.6.0](https://github.com/lbell/pretty-google-calendar/compare/v1.4.1...v1.6.0) and my best guess would be that the issue has something to do with the changes that were made to the AJAX request and/or the event listener that appears to fire it. Obviously, you'd know much better so if you'd like any additional information, or if you'd like me to try any other solutions, let me know.

## Site Info
It's also very possible that the theme or one of the other plugins is not playing nice with this, so I've included a list below.
WordPress Version: 6.3.1
Theme:
![image](https://github.com/lbell/pretty-google-calendar/assets/43394783/99663fa1-fb66-4be4-a907-03d4177d5ca6)
Plugins:
![image](https://github.com/lbell/pretty-google-calendar/assets/43394783/304a3217-ca9c-4853-9b5b-c51953fc98f7)

For now, I'm going to leave my site on version 1.4.1, but once you think you think you may have found the problem, I'll happily test it out for you.

**Comment by lbell** [comment]
Interesting... I'm not having the same issue on any of my sites. Just pushed a 1.6.1 release... maybe try that?

**Comment by lucasbacchi** [comment]
Nope, still broken. I tried 1.6.2 and I have the same issue. Would it be helpful for me to leave it in the broken state? I'm going to revert back to 1.4.1 for now, but I can leave it on the 1.6.2 release for a few days if that would be useful.
The issue is occurring on [this page](https://southchurch.com/our-event#calendar) at the bottom.

This is how I'm embedding it onto the page:
![image](https://github.com/lbell/pretty-google-calendar/assets/43394783/ac02b453-5629-4ebf-8dc2-1a24145cdb28)

Am I doing something wrong?

**Comment by lbell** [comment]
Not sure when I can take a deep dive into this... Don't suppose you could clone the site somewhere at a private URL and leave 1.6.2 on it?

**Comment by mwguerra** [comment]
Hi everyone,

I've encountered this issue at times as well. Clearing the cache usually helps in most cases.

Upon reviewing the code and conducting some tests, I've noticed that even though the pgcal script loads at the footer, the DOM is not always fully loaded when the script executes. This seems to depend on the behavior of other scripts, such as Hotjar, GTM, themes, etc.

One notable difference between the versions discussed here is that the `document.addEventListener("DOMContentLoaded", …)` on pgcal.js has been removed during code cleanup.

Therefore, I suggest re-embedding the code within the DOMContentLoaded event to ensure that the FullCalendar (FC) loads after the DOM has completely loaded. If needed, I can create and send a pull request. Please let me know if that works for you.

Additionally, I've experienced some inconsistent delays with the rendering of the FullCalendar. I'm currently uncertain whether this issue is related to FC itself or a call to the Google API. I will open a new issue once I have gathered more information.

Happy Halloween!

**Comment by lbell** [comment]
@mwguerra: Thanks for sleuthing! That makes sense. Would definitely welcome a pull request if you have a spare moment.
